examples/with-remix-thirdpartyemailpassword/app/routes/_index.tsx

```diff
@@ -0,0 +1,203 @@
+import {
+  CelebrateIcon,
+  SeparatorLine,
+  BlogsIcon,
+  GuideIcon,
+  SignOutIcon,
+} from "../../assets/images";
+import { signOut } from "supertokens-auth-react/recipe/thirdpartyemailpassword/index.js";
+import { recipeDetails } from "../config/frontend";
+import SuperTokens from "supertokens-auth-react";
+import { LoaderFunctionArgs, redirect } from "@remix-run/node";
+import {
+  PreParsedRequest,
+  CollectingResponse,
+} from "supertokens-node/framework/custom/index.js";
+import Session from "supertokens-node/lib/build/recipe/session/index.js";
+import { HTTPMethod } from "supertokens-node/types";
+import { useLoaderData } from "@remix-run/react";
+import { ExtendedSession, SessionDataForUI } from "../lib/superTokensTypes";
+import { SessionContainerInterface } from "supertokens-node/lib/build/recipe/session/types";
+
+function getCookieFromRequest(request: Request) {
+  const cookies: Record<string, string> = {};
+  const cookieHeader = request.headers.get("Cookie");
+  if (cookieHeader) {
+    const cookieStrings = cookieHeader.split(";");
+    for (const cookieString of cookieStrings) {
+      const [name, value] = cookieString.trim().split("=");
+      cookies[name] = value;
+    }
+  }
+  return cookies;
+}
+
+function getQueryFromRequest(request: Request) {
+  const query: Record<string, string> = {};
+  const url = new URL(request.url);
+  const searchParams = url.searchParams;
+  searchParams.forEach((value, key) => {
+    query[key] = value;
+  });
+  return query;
+}
+
+function createPreParsedRequest(request: Request): PreParsedRequest {
+  return new PreParsedRequest({
+    cookies: getCookieFromRequest(request),
+    url: request.url as string,
+    method: request.method as HTTPMethod,
+    query: getQueryFromRequest(request),
+    headers: request.headers,
+    getFormBody: async () => {
+      return await request.formData();
+    },
+    getJSONBody: async () => {
+      return await request.json();
+    },
+  });
+}
+
+export async function loader({
```

Comment:
this should be same as the https://github.com/supertokens/supertokens-node/blob/master/lib/ts/nextjs.ts#L142

Reply:
The functionality you mentioned has been incorporated into the `superTokensHelpers.ts`, in the `getSessionDetails` function, which is called from the loader.